### ISSUE TITLE
Set default expiration time for payment requests 'Never"

### ISSUE THREAD

**Issue by gits7r** [spesmilo/electrum#6364]
Why a fresh wallet will default payment requests expiration to some other value than "Never"?
After upgrading, I had to configure an existing wallet to "Never" on requests tab. I think this should be the default, having 10 mins or something like this will increase the gap of users too much if many requests get burned  / don't end up paid. 

If someone wants a limited lifetime for the validity of a payment request for sure they will set it, but the average users almost never need or want to have an expiring payment request, so I think we should default to "Never" here on both Android and desktop.

**Comment by ecdsa** [comment]
the reasoning is that if onchain invoices expire, the corresponding bitcoin addresses can be freed up and used in new requests without user intervention

**Comment by danuker** [comment]
I agree that it should expire; but perhaps increase the default value to 2 weeks, like [Bitcoin's default mempool expiry](https://github.com/bitcoin/bitcoin/blob/e9fc8f6e7f572224d2bd4577eee6d538331ef663/test/functional/mempool_expiry.py#L22).

See why in [my comment here](https://github.com/spesmilo/electrum/pull/6375#issuecomment-667681991).
